### PR TITLE
Extract import municipalities

### PR DESCRIPTION
**Pull request by koenmetsu**

### CODE REVIEW

File: src/RoadRegistry.BackOffice/Core/GeometryTranslator.cs [hunk 1]
```diff
@@ -4,6 +4,8 @@ namespace RoadRegistry.BackOffice.Core
     using System.Collections.Generic;
     using System.Linq;
     using Be.Vlaanderen.Basisregisters.Shaperon.Geometries;
+    using Messages;
+    using NetTopologySuite.Geometries;
```
Comment (yreynhout) [inline src/RoadRegistry.BackOffice/Core/GeometryTranslator.cs:8]: Please get rid of this by fully qualifying the types in this file.

File: src/RoadRegistry.BackOffice/Messages/Polygon.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace RoadRegistry.BackOffice.Messages
+{
+    public class Polygon
+    {
+        public Ring Ring { get; set; }
```
Comment (yreynhout) [inline src/RoadRegistry.BackOffice/Messages/Polygon.cs:5]: Would you prefer the word `Shell` here? I'm fine with this as well, just not sure what "common" word is used for it.

File: src/RoadRegistry.Editor.ProjectionHost/Program.cs [hunk 3]
```diff
@@ -179,12 +179,15 @@ public static async Task Main(string[] args)
...
-                                            options => options.EnableRetryOnFailure()
+                                            options => options
+                                                .EnableRetryOnFailure()
+                                                .UseNetTopologySuite()
                                         ).Options)
                         )
                         .AddSingleton(sp => new ConnectedProjection<EditorContext>[]
                         {
                             new OrganizationRecordProjection(sp.GetRequiredService<RecyclableMemoryStreamManager>(), WindowsAnsiEncoding),
+                            new MunicipalityGeometryRecordProjection(),
```
Comment (yreynhout) [inline src/RoadRegistry.Editor.ProjectionHost/Program.cs:190]: Doubt we need to use the word record here - `record` is used to denote something that will appear as a record in a dbase / shape file, which is not the case here.

File: src/RoadRegistry.Editor.Schema/Migrations/EditorContextModelSnapshot.cs [hunk 4]
```diff
@@ -54,6 +55,21 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.ToTable("GradeSeparatedJunction","RoadRegistryEditor");
                 });
 
+            modelBuilder.Entity("RoadRegistry.Editor.Schema.MunicipalityGeometryRecord", b =>
+                {
+                    b.Property<string>("NisCode")
+                        .HasColumnType("nvarchar(450)");
```
Comment (yreynhout) [inline src/RoadRegistry.Editor.Schema/Migrations/EditorContextModelSnapshot.cs:61]: I think NISCodes for municipalities are a lot shorter - https://nl.wikipedia.org/wiki/NIS-code. Where is 450 coming from?

File: src/RoadRegistry.Editor.Schema/Migrations/EditorContextModelSnapshot.cs [hunk 5]
```diff
@@ -54,6 +55,21 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+
+                    b.Property<Geometry>("Geometry")
+                        .IsRequired()
+                        .HasColumnType("Geometry");
+
+                    b.HasKey("NisCode")
+                        .HasAnnotation("SqlServer:Clustered", false);
```
Comment (yreynhout) [inline src/RoadRegistry.Editor.Schema/Migrations/EditorContextModelSnapshot.cs:68]: Why would it not be a clustered index?

File: src/RoadRegistry.Editor.Schema/MunicipalityGeometryConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+    public class MunicipalityGeometryConfiguration : IEntityTypeConfiguration<MunicipalityGeometryRecord>
+    {
+        public const string TableName = "MunicipalityGeometry";
+
+        public void Configure(EntityTypeBuilder<MunicipalityGeometryRecord> b)
+        {
+            b.ToTable(TableName, WellknownSchemas.EditorSchema)
```
Comment (yreynhout) [inline src/RoadRegistry.Editor.Schema/MunicipalityGeometryConfiguration.cs:12]: Ah, I see - this is EF being "clever".

File: src/RoadRegistry.Legacy.Extract/Readers/ImportedMunicipalitiesReader.cs [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    using System.Collections.Generic;
+    using BackOffice.Framework;
+    using BackOffice.Messages;
+    using Be.Vlaanderen.Basisregisters.Shaperon;
+    using Be.Vlaanderen.Basisregisters.Shaperon.Geometries;
+    using Microsoft.Data.SqlClient;
+    using Microsoft.Extensions.Logging;
+    using NodaTime;
+    using NodaTime.Text;
+    using Point = BackOffice.Messages.Point;
```
Comment (yreynhout) [inline src/RoadRegistry.Legacy.Extract/Readers/ImportedMunicipalitiesReader.cs:13]: Please remove this and favour full qualification - it makes it more readable and explicit at the location the type is used.

File: src/RoadRegistry.Legacy.Extract/Readers/ImportedMunicipalitiesReader.cs [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    using Polygon = BackOffice.Messages.Polygon;
```
Comment (yreynhout) [inline src/RoadRegistry.Legacy.Extract/Readers/ImportedMunicipalitiesReader.cs:14]: Same here obviously

File: test/RoadRegistry.Tests/BackOffice/SharedCustomizations.cs [hunk 9]
```diff
@@ -4,12 +4,14 @@ namespace RoadRegistry.BackOffice
...
     using AutoFixture;
     using AutoFixture.Dsl;
+    using Be.Vlaanderen.Basisregisters.Shaperon;
     using Be.Vlaanderen.Basisregisters.Shaperon.Geometries;
     using Core;
     using Messages;
     using NetTopologySuite.Geometries;
     using NetTopologySuite.Geometries.Implementation;
     using Uploads;
+    using Polygon = Messages.Polygon;
```
Comment (yreynhout) [inline test/RoadRegistry.Tests/BackOffice/SharedCustomizations.cs:14]: Same here

**Review by yreynhout — COMMENTED** [review]
Looking good overall, just a few cosmetic changes requested.

File: src/RoadRegistry.BackOffice/Messages/Polygon.cs (hunk 2 above)
Comment (koenmetsu) [reply src/RoadRegistry.BackOffice/Messages/Polygon.cs:5]: Shell seems more explicit, let's go for that.

File: src/RoadRegistry.Editor.ProjectionHost/Program.cs (hunk 3 above)
Comment (koenmetsu) [reply src/RoadRegistry.Editor.ProjectionHost/Program.cs:190]: Ah, thanks for the clarification, I didn't get when to use record or not.

File: src/RoadRegistry.Editor.Schema/MunicipalityGeometryConfiguration.cs (hunk 6 above)
Comment (koenmetsu) [reply src/RoadRegistry.Editor.Schema/MunicipalityGeometryConfiguration.cs:12]: What do you mean?

**Review by yreynhout — APPROVED** [review]

**Pull request merged**